packages/app-shell-ui/src/utils/filterModel.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  HvAppShellHeaderActionModel,
+  HvAppShellMenuModel,
+  HvAppShellModel,
+  HvAppShellProvidersModel,
+  HvAppShellServiceModel,
+  HvAppShellServicesModel,
+  HvAppShellViewModel,
+} from "@hitachivantara/app-shell-shared";
+
+import type { ConditionResultsList } from "../hooks/useConditionsEvaluator";
+
+const shouldInclude = (
+  conditions: Array<{ globalIndex?: number } | Record<string, any>> | undefined,
+  conditionResults: ConditionResultsList,
+): boolean => {
+  if (!conditions || conditions.length === 0) {
+    return true;
+  }
+
+  for (const condition of conditions) {
+    const idx =
+      condition && typeof condition.globalIndex === "number"
```

Comment:
All conditions should necessarily have a defined, numeric globalIndex property. No?